app/src/main/java/com/alexbar/layoutbasic/screen/SnakeGameScreen.kt

```diff
@@ -0,0 +1,128 @@
+package com.alexbar.layoutbasic.screen
+
+import androidx.compose.foundation.Canvas
+import androidx.compose.foundation.gestures.detectTapGestures
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.material3.Text
+import androidx.compose.runtime.*
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.geometry.*
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.drawscope.DrawScope
+import androidx.compose.ui.input.pointer.pointerInput
+import androidx.compose.ui.tooling.preview.Preview
+import com.alexbar.layoutbasic.ui.theme.*
+import com.alexbar.layoutbasic.utils.GameConstants
+import kotlinx.coroutines.delay
+import kotlin.random.Random
+
+@Composable
+fun SnakeGame() {
+    val snakeBodyColor = Color.Blue
+    val foodColor = Color.Red
+    val snakeSize = Dimens.snake_size_100
+    var snakePosition by remember { mutableStateOf(Offset.Zero) }
+    var direction by remember { mutableStateOf(Offset(snakeSize, Dimens.snake_init_direction_y)) }
+    var foodPosition by remember { mutableStateOf(randomOffset(snakeSize)) }
+    var gameOver by remember { mutableStateOf(false) }
+
+    LaunchedEffect(key1 = true) {
+        while (!gameOver) {
+            delay(GameConstants.game_delay_300_ms)
+            val newPosition = snakePosition + direction
+            if (newPosition.x < 0 || newPosition.y < 0) {
+                gameOver = true
+            } else {
+                snakePosition = newPosition
+            }
+
+            if (snakePosition == foodPosition) {
+                foodPosition = randomOffset(snakeSize)
+            }
+        }
+    }
+
+    if (!gameOver) {
+        Canvas(modifier = Modifier
+            .fillMaxSize()
+            .pointerInput(Unit) {
+                detectTapGestures(
+                    onTap = { offset ->
+                        direction = changeDirection(snakePosition, offset, snakeSize)
+                    }
+                )
+            }
+        ) {
+
+            drawBoard(snakeSize, Dimens.board_width, Dimens.board_height)
+            drawSnake(snakePosition, snakeBodyColor)
+            drawCell(foodColor, foodPosition, Size(snakeSize, snakeSize))
+        }
+    } else {
+        Text(
+            text = GameConstants.game_over_text,
+            color = Color.Black,
+            modifier = Modifier.fillMaxSize()
+        )
+    }
+}
+
+fun DrawScope.drawBoard(
+    snakeSize: Float,
+    boardWidth: Int,
+    boardHeight: Int
+) {
+    for (i in 0 until boardWidth) {
+        for (j in 0 until boardHeight) {
+            val isDarkCell = (i + j) % 2 == 1
+            val cellColor = if (isDarkCell) BoardDarkColor else BoardLightColor
+            val cellTopLeft = Offset(i * snakeSize, j * snakeSize)
+            drawCell(cellColor, cellTopLeft, Size(snakeSize, snakeSize))
+        }
+    }
+}
```

Comment:
@ezatpanah How can I implement this dynamically? I set these values as constants (board_width = 11, board_height = 23) because they functioned well in the simulator I was using for testing, but I assume they might not display correctly on phones with larger or smaller screens.

Reply:
This is my changes for having this method dynamically :

`fun DrawScope.drawBoard(
    snakeSize: Float,
    screenWidth: Float,
    screenHeight: Float
) {
    val cellCountX = (screenWidth / snakeSize).toInt()
    val cellCountY = (screenHeight / snakeSize).toInt()

    for (i in 0 until cellCountX) {
        for (j in 0 until cellCountY) {
            val isDarkCell = (i + j) % 2 == 1
            val cellColor = if (isDarkCell) BoardDarkColor else BoardLightColor
            val cellTopLeft = Offset(i * snakeSize, j * snakeSize)
            drawCell(cellColor, cellTopLeft, Size(snakeSize, snakeSize))
        }
    }
}
`

and then :
you can call this method like :
`            drawBoard(snakeSize, size.width, size.height)
`

in this changes :
instead of specifying boardWidth and boardHeight as constants, you can pass the screen dimensions (screenWidth and screenHeight) to the drawBoard function. This way, the board will be dynamically drawn based on the screen size.

Reply:
@ezatpanah When calling the method drawBoard, where does `size` come from? I discovered that I could use `LocalConfiguration.current.screenWidthDp` and `LocalConfiguration.current.screenHeightDp`, but when I tested the code, it didn’t work properly because it renders the board too small. Additionally, I’m wondering what units should I use for these values, should they be in Float or in pixels?

Reply:
the size provides the dimensions of the current drawing environment 
even this way as well works for your code :

`
fun DrawScope.drawBoard(
    snakeSize: Float,
    size: Size
) {
    val boardWidth = (size.width / snakeSize).toInt()
    val boardHeight = (size.height / snakeSize).toInt()

    for (i in 0 until boardWidth) {
        for (j in 0 until boardHeight) {
            val isDarkCell = (i + j) % 2 == 1
            val cellColor = if (isDarkCell) BoardDarkColor else BoardLightColor
            val cellTopLeft = Offset(i * snakeSize, j * snakeSize)
            drawCell(cellColor, cellTopLeft, Size(snakeSize, snakeSize))
        }
    }
}

`
and then in your Canvas :

`
    Canvas(
        modifier = Modifier.fillMaxSize(),
        onDraw = { drawScope: DrawScope ->
...
            val availableSize = this.size // Use the size provided by the Canvas
            drawScope.drawBoard(snakeSize, availableSize)
...
        }
    )
`

please check this part of code and let me know , is it correct or not